doc/cephadm/services/monitoring.rst

```diff
@@ -156,6 +156,38 @@ example spec file:
 
 .. _cephadm_monitoring-images:
 
+.. _cephadm_default_images:
+
+Default images
+~~~~~~~~~~~~~~
+
+*The information in this section was developed by Eugen Block in a thread on
+the [ceph-users] mailing list in April of 2024. The thread can be viewed here:
+``https://lists.ceph.io/hyperkitty/list/ceph-users@ceph.io/thread/QGC66QIFBKRTPZAQMQEYFXOGZJ7RLWBN/``.*
+
+``cephadm`` stores a local copy of the ``cephadm`` binary in
+``var/lib/ceph/{FSID}/cephadm.{DIGEST}``, where ``{DIGEST}`` is an alphanumeric
+string representing the currently-running version of Ceph.
+
+To see the default container images, run a command of the following form:
+
+.. prompt:: bash #
+
+   grep -E "DEFAULT*IMAGE" /var/lib/ceph/{FSID}/cephadm.{DIGEST}
```

Comment:
I think this only works in pacific and quincy, where the cephadm isn't a zip file. In main/squid, I found this process works instead
```
root@smithi027:~# mkdir cephadm_dir
root@smithi027:~# 
root@smithi027:~# unzip /var/lib/ceph/b9943a26-07e0-11ef-bc96-c7b262605968/cephadm.8a4210adf49bf7cbbb308ea609b0f309661d65fe037af8a6095519adb47c62fe -d cephadm_dir > /dev/ull
warning [/var/lib/ceph/b9943a26-07e0-11ef-bc96-c7b262605968/cephadm.8a4210adf49bf7cbbb308ea609b0f309661d65fe037af8a6095519adb47c62fe]:  14 extra bytes at beginning or within zipfile
  (attempting to process anyway)
root@smithi027:~# 
root@smithi027:~# egrep "_IMAGE" cephadm_dir/cephadmlib/constants.py
DEFAULT_IMAGE = 'quay.ceph.io/ceph-ci/ceph:main'
DEFAULT_IMAGE_IS_MAIN = True
DEFAULT_IMAGE_RELEASE = 'squid'
DEFAULT_PROMETHEUS_IMAGE = 'quay.io/prometheus/prometheus:v2.43.0'
DEFAULT_LOKI_IMAGE = 'docker.io/grafana/loki:2.4.0'
DEFAULT_PROMTAIL_IMAGE = 'docker.io/grafana/promtail:2.4.0'
DEFAULT_NODE_EXPORTER_IMAGE = 'quay.io/prometheus/node-exporter:v1.5.0'
DEFAULT_ALERT_MANAGER_IMAGE = 'quay.io/prometheus/alertmanager:v0.25.0'
DEFAULT_GRAFANA_IMAGE = 'quay.io/ceph/grafana:9.4.12'
DEFAULT_HAPROXY_IMAGE = 'quay.io/ceph/haproxy:2.3'
DEFAULT_KEEPALIVED_IMAGE = 'quay.io/ceph/keepalived:2.2.4'
DEFAULT_NVMEOF_IMAGE = 'quay.io/ceph/nvmeof:1.2.1'
DEFAULT_SNMP_GATEWAY_IMAGE = 'docker.io/maxwo/snmp-notifier:v1.2.1'
DEFAULT_ELASTICSEARCH_IMAGE = 'quay.io/omrizeneva/elasticsearch:6.8.23'
DEFAULT_JAEGER_COLLECTOR_IMAGE = 'quay.io/jaegertracing/jaeger-collector:1.29'
DEFAULT_JAEGER_AGENT_IMAGE = 'quay.io/jaegertracing/jaeger-agent:1.29'
DEFAULT_JAEGER_QUERY_IMAGE = 'quay.io/jaegertracing/jaeger-query:1.29'
DEFAULT_SMB_IMAGE = 'quay.io/samba.org/samba-server:devbuilds-centos-amd64'
```

on reef the process is similar, although you would run the grep against `cephadm_dir/__main__.py` rather than `cephadm_dir/cephadmlib/constants.py`

On pacific and quincy, the procedure you have here I think works, although I couldn't get this specific `grep -E DEFAULT*IMAGE` to work for me. I had better luck with `grep -E "_IMAGE"

```
root@smithi027:~# grep -E "DEFAULT*IMAGE" cephadm_dir/cephadmlib/constants.py
root@smithi027:~# 
root@smithi027:~# grep -E "_IMAGE" cephadm_dir/cephadmlib/constants.py
DEFAULT_IMAGE = 'quay.ceph.io/ceph-ci/ceph:main'
DEFAULT_IMAGE_IS_MAIN = True
DEFAULT_IMAGE_RELEASE = 'squid'
DEFAULT_PROMETHEUS_IMAGE = 'quay.io/prometheus/prometheus:v2.43.0'
DEFAULT_LOKI_IMAGE = 'docker.io/grafana/loki:2.4.0'
DEFAULT_PROMTAIL_IMAGE = 'docker.io/grafana/promtail:2.4.0'
DEFAULT_NODE_EXPORTER_IMAGE = 'quay.io/prometheus/node-exporter:v1.5.0'
DEFAULT_ALERT_MANAGER_IMAGE = 'quay.io/prometheus/alertmanager:v0.25.0'
DEFAULT_GRAFANA_IMAGE = 'quay.io/ceph/grafana:9.4.12'
DEFAULT_HAPROXY_IMAGE = 'quay.io/ceph/haproxy:2.3'
DEFAULT_KEEPALIVED_IMAGE = 'quay.io/ceph/keepalived:2.2.4'
DEFAULT_NVMEOF_IMAGE = 'quay.io/ceph/nvmeof:1.2.1'
DEFAULT_SNMP_GATEWAY_IMAGE = 'docker.io/maxwo/snmp-notifier:v1.2.1'
DEFAULT_ELASTICSEARCH_IMAGE = 'quay.io/omrizeneva/elasticsearch:6.8.23'
DEFAULT_JAEGER_COLLECTOR_IMAGE = 'quay.io/jaegertracing/jaeger-collector:1.29'
DEFAULT_JAEGER_AGENT_IMAGE = 'quay.io/jaegertracing/jaeger-agent:1.29'
DEFAULT_JAEGER_QUERY_IMAGE = 'quay.io/jaegertracing/jaeger-query:1.29'
DEFAULT_SMB_IMAGE = 'quay.io/samba.org/samba-server:devbuilds-centos-amd64'
```

Reply:
Just remembered this is a backport. We'd need to make a change in main as well to account for this. Do we want to merge this as is and then apply a patch to main and bring it down, or make an adjustment here before merging @zdover23 ? If the former, feel free to dismiss my review so you can just merge this.

Reply:
Thanks for this @adk3798. I'll make these alterations (and add material to all the current release branches to explain these differences) later in the day.

Reply:
> Just remembered this is a backport. We'd need to make a change in main as well to account for this. Do we want to merge this as is and then apply a patch to main and bring it down, or make an adjustment here before merging @zdover23 ? If the former, feel free to dismiss my review so you can just merge this.

I'll merge it and then make the alterations you suggested. The docs won't display this inaccuracy for more than twelve hours. Fortunately, the docs aren't as hard-nosed as the code, so they can afford to be a palimpsest.

Reply:
Works for me. Dismissed by review

Reply:
https://github.com/ceph/ceph/pull/57225 - Squid procedure

Reply:
https://github.com/ceph/ceph/pull/57236 - Reef procedure

Reply:
Jostling my way into this nearly cold discussion:  Adam's procedure will work but I don't think the dev team is going to guarantee that these constants will exist in that location indefinitely. 

Personally, I would like to see the code make it simpler to accomplish the task of listing images (default and current) and making it easier to mirror images... but that's a task for the future.

Reply:
@phlogistonjohn, I'll keep abreast of this and make sure that the docs are as current as the shifting sands of technology permit.